android/app/src/main/java/com/sample/react/TestAppReactNativeHost.kt

```diff
@@ -2,45 +2,69 @@ package com.sample.react
 
 import android.app.Activity
 import android.app.Application
-import com.facebook.react.*
+import com.facebook.react.PackageList
+import com.facebook.react.ReactInstanceManager
+import com.facebook.react.ReactNativeHost
+import com.facebook.react.ReactPackage
 import com.facebook.react.bridge.ReactMarker
 import com.facebook.react.bridge.ReactMarkerConstants
 import com.facebook.react.common.LifecycleState
 import com.facebook.soloader.SoLoader
 import javax.inject.Inject
 import javax.inject.Singleton
 
+sealed class BundleSource {
+    enum class Action {
+        RESTART, RELOAD
+    }
+
+    abstract fun moveTo(to: BundleSource): Action
+
+    object Disk : BundleSource() {
+        override fun moveTo(to: BundleSource) = Action.RESTART
```

Comment:
IIRC, it should be sufficient to do `Action.RELOAD` for Disk -> Server.

Reply:
I believe that when you switch to loading from the server, you have to rebuild the `ReactInstanceManager` with the correct `useDeveloperSupport` flag set and the asset name.

Reply:
In other words, I would leave it as it is.